src/main/java/com/microsoft/azure/functions/middleware/MiddlewareExecutionContext.java

```diff
@@ -0,0 +1,22 @@
+package com.microsoft.azure.functions.middleware;
+
+import com.microsoft.azure.functions.ExecutionContext;
+
+/**
+ * Expands {@link ExecutionContext} so that {@link FunctionWorkerMiddleware} implementations
+ * can modify target function instance or get more information about the target function.
+ */
+public interface MiddlewareExecutionContext extends ExecutionContext {
+    public ClassLoader getFunctionClassLoader();
+    public Class getFunctionClass();
+
+    public Object getFunctionInstance();
```

Comment:
For Quarkus, we don't need `getFunctionClassLoader` and `getFunctionInstance` right. Please correct me if I am wrong.

Reply:
getFunctionClassLoader() is removable so long as getFunctionClass().getClassLoader() returns the correct loader.

getFunctionInstance() you'd want to have to see if another middleware has overriden the instance already and proxy that instance, output a warning, or abort.  An alternative to that is for setFunctionInstance() to throw an exception if the instance has already been set.  What do you think is the best approach?

Reply:
Make sense, I was just wondering if we would have such a use case. The only middleware that require access to FunctionInstance will be DI framework like Quarkus or Spring, etc. But cx will not use both Quarkus and Spring at the same time. 
However, I can see this is valid point. I will explore this more from our side.

Reply:
middleware would be pulled in via a maven dependency.  So, if an app developer accidently pulled in both a Quarkus and Spring middleware dependency by accident (or on purpose) you could have conflict.  These sort of dependency conflicts are more common than you think based on my experience.

Not that important though.  Just something to think about.